### ISSUE TITLE
Istio cni configmap namespace should be configurable

### ISSUE THREAD

**Issue by aniekgul** [kiali/kiali#6076]
https://github.com/kiali/kiali/blob/master/business/istio_config.go#L1204

Kiali currently looks for the istio cni configmap in the `istio-system` namespace however the istio cni installation may reside in a different namespace such as in environments where the `system-node-critical` priority class is restricted to the `kube-system` namespace only.

**Comment by jshaughn** [comment]
Kiali actually looks in the configured control plane namespace, which may or may not be `istio-system`, but I guess the problem is the same.  I think you are saying that the configMap could be located in a different namespace and may not be allowed to be located in the control plane namespace.  Correct?

**Comment by aniekgul** [comment]
Yep exactly, the configmap could be in a different namespace than the control plane.

**Comment by jmazzitelli** [comment]
I would prefer to figure out a better way (if there is one) to detect whether ambient is enabled or not.

It would suck if we had to add yet one more "customizable Istio configuration object name/location" setting in the Kiali CR. Hopefully there is a different way we can figure out if ambient is enabled or not.

We should ask the Istio folks if there is a better, or if looking at this ConfigMap is the only way.

**Comment by josunect** [comment]
> @aniekgul., I think this is brand new code we're using to determine whether Ambient is enabled. I guess it's the first time we're peeking into the istio-cni-config map. I'm not exactly sure if this is even the final approach to detecting Ambient, but thanks for pointing this out.
> 
> cc @josunect , if we think this is the best way of detecting ambient, we may need a new config entry. What do you think?

Yes, that's right, it is to check if Ambient is enabled. 

I haven't see information in the isito config map, but I've found it in the istio-cni-config config map. 

I can see that maybe inspecting the istiod pod can have some information, or getting the profile - As this is installed with profile=ambient, but I've thought the config-map would be easier and more reliable. 

But I agree, @jmazzitelli , we can ask what would be the best way.

**Comment by josunect** [comment]
Regarding to that thread, the idea is that when Ambient is stable all the Ambient components will be installed by default, so we would just keep watching the specific pods to know what components (sidecar, ztunnel or waypoint) are using, and not checking if Ambient is enabled globally. 

IMO it is interesting to know in Kiali, at the moment, if Istio is in Ambient profile or not, to keep having the same behavior if Ambient is not enabled - And there are not lots of changes, it is basically the message _Missing sidecars_ VS _Out of mesh_, but it could be confusing anyway. Probably it is limited at the moment, as it cannot work if the istio-cni-config is not in the istio-system namespace, but not sure if it will worth the effort to make a configuration item for this, as it will be probably temporary. So maybe we could look for other alternatives - like what kind of components are installed in the istio namespace?

**Comment by jshaughn** [comment]
@aniekgul, @josunect , I'm going to close this as a "Wont Do",  Not because it isn't valid but because this is only for Ambient support, which is experimental and as it matures this global profile will likely go away, meaning we won't need to make this check.  Instead, we will likely focus on namespace and object-level Ambient impact.

**Comment by jmazzitelli** [comment]
See the slack thread for more details regarding this decision. https://istio.slack.com/archives/C041EQL1XMY/p1682674995416059
